python/lsst/image_cutout_backend/stencils.py

```diff
@@ -0,0 +1,487 @@
+        """
+        factor = (2 * np.pi / n_vertices) * radians
+        return SkyPolygon(self._center.offset(b * factor, self._radius) for b in range(n_vertices))
+
+    def to_pixels(self, wcs: SkyWcs, bbox: Box2I) -> PixelStencil:
+        # Docstring inherited.
+        transform = makeWcsPairTransform(_make_local_gnomonic_wcs(self._center), wcs)
+        affine = linearizeTransform(transform, Point2D(0.0, 0.0))
+        sky_ellipse_core = lsst.afw.geom.ellipses.Axes(
+            self._radius.asRadians(), self._radius.asRadians(), 0.0
+        )
+        pixel_ellipse_core = sky_ellipse_core.transform(affine.getLinear())
+        spans = SpanSet.fromShape(
+            lsst.afw.geom.ellipses.Ellipse(pixel_ellipse_core, wcs.skyToPixel(self._center))
+        )
+        if not bbox.contains(spans.getBBox()):
+            if self._clip:
+                spans = spans.clippedTo(bbox)
+            else:
+                raise StencilNotContainedError(
+                    f"{self} has bbox {spans.getBBox()} in pixel coordinates, which is not within {bbox}."
+                )
+        return SpanSetStencil(spans)
+
+    @property
+    def region(self) -> lsst.sphgeom.Region:
+        # Docstring inherited.
+        return lsst.sphgeom.Circle(
+            self._center.getVector(), lsst.sphgeom.Angle.fromRadians(self._radius.asRadians())
+        )
+
+    def to_fits_metadata(self, metadata: PropertyList) -> None:
+        # Docstring inherited.
+        metadata.set("ST_TYPE", "CIRCLE", "Type of stencil used to create this cutout.")
+        metadata.set("ST_RA", self._center.getRa().asDegrees(), "Circle center right ascension in degrees.")
+        metadata.set("ST_DEC", self._center.getDec().asDegrees(), "Circle center declination in degrees.")
+        metadata.set("ST_RAD", self._radius.asDegrees(), "Circle radius in degrees.")
+
+    @property
+    def fingerprint(self) -> bytes:
+        # Docstring inherited.
+        hasher = blake2b(digest_size=16)
+        hasher.update(b"CIRCLE")
+        hasher.update(struct.pack("!d", self._center.getRa().asRadians()))
+        hasher.update(struct.pack("!d", self._center.getDec().asRadians()))
+        hasher.update(struct.pack("!d", self._radius.asRadians()))
+        return hasher.digest()
+
+
+class SkyPolygon(SkyStencil):
+    """A sky-coordinate stencil in the shape of a great-circle polygon.
+
+    Parameters
+    ----------
+    vertices : `Iterable` [ `SpherePoint` ]
+        Vertices of the polygon, CCW when looking out from the origin.
+        Implicitly closed (the first vertex should not be duplicated as the
+        last).
+    clip : `bool`, optional
+        If `True` (`False` is default), clip pixel stencils returned by
+        `to_pixels` instead of raising `StencilNotContainedError`.
+
+    Notes
+    -----
+    Vertex orientation is not checked at construction, and incorrect
+    orientation may result in unspecified failures in `to_pixels`.
```

Comment:
I assume this will be fixed in a future version?

Reply:
Yes.  It'll require some moderately tricky geometry code that I'm not looking forward to writing, but it's clear we need to do it; I just drew the boundary for this ticket at things I could do with our existing low-level geometry code.